doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,430 @@
+
+The Zephyr CMake package search functionality allows for explicitly specifying
+a Zephyr base using an environment variable.
+
+To do this, use the following ``find_package()`` syntax:
+
+.. code-block:: cmake
+
+   find_package(Zephyr HINTS $ENV{ZEPHYR_BASE})
+
+This syntax instructs CMake to first search for Zephyr using the Zephyr base environment setting
+:envvar:`ZEPHYR_BASE` and then use the normal search paths.
+
+.. _zephyr_cmake_search_order:
+
+Zephyr CMake Package Search Order
+*********************************
+
+When Zephyr base environment setting is not used for searching, the Zephyr installation matching
+the following criteria will be used:
+
+* A Zephyr repository application will use the Zephyr in which it is located.
+  For example:
+
+  .. code-block:: none
+
+        <projects>/zephyr-workspace/zephyr
+        └── samples
+            └── hello_world
+
+  in this example, ``hello_world`` will use ``<projects>/zephyr-workspace/zephyr``.
+
+
+* Zephyr workspace application will use the Zephyr that share the same workspace.
+  For example:
+
+  .. code-block:: none
+
+     <projects>/zephyr-workspace
+     ├── zephyr
+     ├── ...
+     └── my_applications
+          └── my_first_app
+
+  in this example, ``my_first_app`` will use ``<projects>/zephyr-workspace/zephyr`` as this Zephyr
+  is located in the same workspace as the Zephyr workspace application.
+
+
+* Zephyr freestanding application will use the Zephyr registered in the CMake user package registry.
+  For example:
+
+  .. code-block:: none
+
+     <projects>/zephyr-workspace-1
+     └── zephyr                       (Not exported to CMake)
+
+     <projects>/zephyr-workspace-2
+     └── zephyr                       (Exported to CMake)
+
+     <home>/app
+     ├── CMakeLists.txt
+     ├── prj.conf
+     └── src
+         └── main.c
+
+  in this example, only ``<projects>/zephyr-workspace-2/zephyr`` is exported to the CMake package
+  registry and therefore this Zephyr will be used by the Zephyr freestanding application
+  ``<home>/app``.
+
+  If user wants to test the application with ``<projects>/zephyr-workspace-1/zephyr``, this can be
+  done by using the Zephyr Base environment setting, meaning set
+  ``ZEPHYR_BASE=<projects>/zephyr-workspace-1/zephyr``, before
+  running CMake.
+
+  .. note::
+
+     The Zephyr package selected on the first CMake invocation will be used for all subsequent
+     builds. To change the Zephyr package, for example to test the application using Zephyr base
+     environment setting, then it is necessary to do a pristine build first
+     (See :ref:`application_rebuild`).
+
+Zephyr CMake Package Version
+****************************
+
+When writing an application then it is possible to specify a Zephyr version number ``x.y.z`` that
+must be used in order to build the application.
+
+Specifying a version is especially useful for a Zephyr freestanding application as it ensures the
+application is built with a minimal Zephyr version.
```

Comment:
The sentence doesn't really give new information, IMO it should be merged with the next sentence.
```suggestion
Specifying a version for a Zephyr freestanding application ensures CMake picks a compatible Zephyr installation
when there are multiple installations in the system.
```

Reply:
Good suggestion.